moveit_ros/perception/handeye_calibration_target/src/handeye_aruco_target.cpp

```diff
@@ -0,0 +1,258 @@
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Yu Yan */
+
+#include <moveit/handeye_calibration_target/handeye_aruco_target.h>
+
+namespace moveit_handeye_calibration
+{
+const std::string LOGNAME = "handeye_aruco_target";
+
+std::vector<std::string> HandEyeArucoTarget::getDictionaryIds()
+{
+  std::vector<std::string> ids;
+  for (const std::pair<const std::string, cv::aruco::PREDEFINED_DICTIONARY_NAME>& name : dict_map_)
+    ids.push_back(name.first);
+  return ids;
+}
+
+bool HandEyeArucoTarget::setTargetIntrinsicParams(const int& markers_x, const int& markers_y, const int& marker_size,
+                                                  const int& separation, const int& border_bits,
+                                                  const std::string& dictionary_id)
+{
+  if (markers_x != markers_x_ || markers_y != markers_y_ || marker_size != marker_size_ || separation != separation_ ||
+      border_bits != border_bits_ || dict_map_.find(dictionary_id)->second != dict_)
+    ROS_DEBUG_STREAM_NAMED(LOGNAME, "Set target intrinsic params: "
+                                        << "\n"
+                                        << "markers_x_ " << std::to_string(markers_x) << "\n"
+                                        << "markers_y_ " << std::to_string(markers_y) << "\n"
+                                        << "marker_size " << std::to_string(marker_size) << "\n"
+                                        << "separation " << std::to_string(separation) << "\n"
+                                        << "border_bits" << std::to_string(border_bits) << "\n"
+                                        << "dictionary_id " << dictionary_id << "\n");
+  if (markers_x > 0 && markers_y > 0 && marker_size > 0 && separation > 0 && border_bits > 0 && !dictionary_id.empty())
+  {
+    std::lock_guard<std::mutex> lck(aruco_mtx_);
+    markers_x_ = markers_x;
+    markers_y_ = markers_y;
+    marker_size_ = marker_size;
+    separation_ = separation;
+    border_bits_ = border_bits;
+
+    const auto& it = dict_map_.find(dictionary_id);
+    if (it != dict_map_.end())
+      dict_ = it->second;
+    else
+    {
+      ROS_ERROR_STREAM_NAMED(LOGNAME, "Invalid dictionary name.");
+      return false;
+    }
+  }
+  else
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Invalid target intrinsic params.");
+    return false;
+  }
+
+  return true;
+}
+
+bool HandEyeArucoTarget::setTargetDimension(const double& marker_size, const double& marker_seperation)
```

Comment:
```suggestion
bool HandEyeArucoTarget::setTargetDimension(const double& marker_size, const double& marker_separation)
```

Ctrl+H for "seperat" -> "separat"

Reply:
The error spellings of "sepraration" are corrected.